src/index/hnsw/impl/IndexWrapperCosine.cc

```diff
@@ -0,0 +1,29 @@
+// Copyright (C) 2019-2024 Zilliz. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance
+// with the License. You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License
+// is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+// or implied. See the License for the specific language governing permissions and limitations under the License.
+
+#include "index/hnsw/impl/IndexWrapperCosine.h"
+
+#include <faiss/IndexCosine.h>
+
+namespace knowhere {
+
+// a wrapper that overrides a distance computer
+IndexWrapperCosine::IndexWrapperCosine(faiss::Index* index, const float* inverse_l2_norms_in)
```

Comment:
Why we need a wrapper instead of putting logics in `IndexHNSWxxxxCosine`?

Reply:
Ultimately, it was needed for refine and to enable/disable refine for the search wirh a config parameter.